amoro-format-mixed/amoro-mixed-flink/amoro-mixed-flink-common/src/main/java/org/apache/amoro/flink/read/internals/KafkaSource.java

```diff
@@ -179,7 +179,7 @@ public SplitEnumerator<KafkaPartitionSplit, KafkaSourceEnumState> restoreEnumera
         props,
         enumContext,
         boundedness,
-        checkpoint.assignedPartitions());
+        checkpoint);
```

Comment:
After flink 1.18, the kakfa function here has changed.
https://github.com/apache/flink-connector-kafka/blame/441e201e90bf3f8f5f9c8d7d72965674f7db8ab4/flink-connector-kafka/src/main/java/org/apache/flink/connector/kafka/source/enumerator/KafkaSourceEnumerator.java#L125C13-L125C57